pkgs/os-specific/linux/intel-sgx/default.nix

```diff
@@ -0,0 +1,286 @@
+
+      cp external/ippcp_internal/inc/ippcp.h $out/include/
+      cp external/ippcp_internal/inc/ippcpdefs.h $out/include/
+      cp external/ippcp_internal/inc/ippversion.h $out/include/
+      cp external/ippcp_internal/inc/sgx_ippcp.h $out/include/
+
+      cp external/dcap_source/QuoteGeneration/quote_wrapper/common/inc/sgx_ql_lib_common.h $out/include/
+      cp external/dcap_source/QuoteGeneration/quote_wrapper/common/inc/sgx_quote_3.h $out/include/
+      cp external/dcap_source/QuoteGeneration/quote_wrapper/common/inc/sgx_ql_quote.h $out/include/
+      cp external/dcap_source/QuoteGeneration/pce_wrapper/inc/sgx_pce.h $out/include/
+      cp external/dcap_source/QuoteVerification/QvE/Include/sgx_qve_header.h $out/include/
+      cp external/dcap_source/QuoteVerification/dcap_tvl/sgx_dcap_tvl.h $out/include/
+      cp external/dcap_source/QuoteVerification/dcap_tvl/sgx_dcap_tvl.edl $out/include/
+
+      cp -r sdk/tlibcxx/include $out/include/libcxx
+
+      # copy sample code
+      cp -r SampleCode $out/SampleCode
+      mkdir $out/SampleCode/RemoteAttestation/sample_libcrypto
+      cp ${build}/libsample_libcrypto.so $out/SampleCode/RemoteAttestation/sample_libcrypto/
+      cp sdk/sample_libcrypto/sample_libcrypto.h $out/SampleCode/RemoteAttestation/sample_libcrypto/
+
+      # copy libraries
+      mkdir -p $out/lib
+      cp ${build}/libsgx_ptrace.so $out/lib/
+      cp ${build}/libsgx_tcrypto.a $out/lib/
+      cp ${build}/libsgx_tkey_exchange.a $out/lib/
+      cp ${build}/libsgx_trts.a $out/lib/
+      cp ${build}/libsgx_trts_sim.a $out/lib/
+      cp ${build}/libsgx_tservice.a $out/lib/
+      cp ${build}/libsgx_tservice_sim.a $out/lib/
+      cp ${build}/libsgx_tstdc.a $out/lib/
+      cp ${build}/libsgx_tcxx.a $out/lib/
+      cp ${build}/libsgx_tcmalloc.a $out/lib/
+      cp ${build}/libsgx_tswitchless.a $out/lib/
+      cp ${build}/libsgx_uswitchless.a $out/lib/
+      cp ${build}/libsgx_ukey_exchange.a $out/lib/
+      cp ${build}/libsgx_capable.a $out/lib/
+      cp ${build}/libsgx_capable.so $out/lib/
+      cp ${build}/libsgx_uprotected_fs.a $out/lib/
+      cp ${build}/libsgx_tprotected_fs.a $out/lib/
+      cp ${build}/libsgx_pcl.a $out/lib/
+      cp ${build}/libsgx_pclsim.a $out/lib/
+      cp ${build}/libsgx_pthread.a $out/lib/
+      cp ${build}/libsgx_omp.a $out/lib/
+
+      cp ${build}/libsgx_epid_deploy.so $out/lib/libsgx_epid.so
+      cp ${build}/libsgx_epid_sim.so $out/lib/
+      cp ${build}/libsgx_launch_deploy.so $out/lib/libsgx_launch.so
+      cp ${build}/libsgx_launch_sim.so $out/lib/
+      cp ${build}/libsgx_quote_ex_deploy.so $out/lib/libsgx_quote_ex.so
+      cp ${build}/libsgx_quote_ex_sim.so $out/lib/
+      cp ${build}/libsgx_uae_service_deploy.so $out/lib/libsgx_uae_service.so
+      cp ${build}/libsgx_uae_service_sim.so $out/lib/
+      cp ${build}/libsgx_urts_deploy.so $out/lib/libsgx_urts.so
+      cp ${build}/libsgx_urts_sim.so $out/lib/libsgx_urts_sim.so
```

Comment:
Can we use wildcards here?

Reply:
No - because some of the libraries have different names in the output directory, so we can't do a blanket copy on them all.